STX.REST.RESTFulSense.Clients/Models/Services/HttpExchanges/Exceptions/FailedHttpExchangeException.cs

```diff
@@ -0,0 +1,16 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using Xeptions;
+
+namespace STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions
+{
+    internal class FailedHttpExchangeException : Xeption
+    {
+        public FailedHttpExchangeException(string message, Exception innerException)
```

Comment:
@glhays this local exception is used to wrap external exceptions. For example, the `HttpRequestException`, `TaskCanceledException` and others which are not a type of the `Xeption` class

Reply:
@aosunlana after further research, I agree with your comment as Xeption is not appropriate here.
Change the exception model name to `FailedHttpExchangeRequestException` In all your test logic this exception model is thrown based on a HttpRequest exception so the name should be appropriate for its purpose. Not another Generic where it may be used by accident where not completely understood by a vague naming. Unless you have a better argument not to change it?

Reply:
This exception "FailedHttpExchangeException" should wrap too any exception thrown when dealing with Content. Maybe I am something wrong with some detail, so I think it has no sense adding the Request word into it.

Reply:
@LBoullosa then that goes against what @aosunlana states previously. Currently this exception is only used in one test, and only is used in a request. Nowhere in any other Standard compliant system have I seen generic "vague" named exception models than what I am starting to see more here. Catch all exceptions?

Reply:
![image](https://github.com/The-Standard-Organization/STX.REST.RESTFulSense.Clients/assets/81052783/212fa51e-99ba-47d8-ba02-383489a473ce)

Reply:
@glhays I used the Exception to wrap the different external exceptions which are not of the type `Xeption`.

Reply:
I agree with @glhays, the local exceptions should have a specific and meaningful name so that it can be appropriately handled for things like logging (log level) or replay where needed.  I would lean to these exception mappings for the external exceptions
`HttpRequestException` maps to `FailedRequestHttpExchangeException`
`TaskCanceledException` maps to `TaskCanceledHttpExchangeException`
`ArgumentException` maps to `InvalidArgumentHttpExchangeException`
`FormatException` maps to `InvalidFormatHttpExchangeException`
`ObjectDisposedException` maps to `ObjectDisposedHttpExchangeException`
`InvalidOperationException` maps to `InvalidOperationHttpExchangeException`

I don't think the generic catch all apply here.  For the catch all, I see it more in the case where you have a bunch of exceptions that inherits from the same base class.  If you need to treat any one of them differently, then you would create a specific local exception for that one, otherwise it can be handled genericly i.e. `DbUpdateConcurrencyException` (which inherists from `DbUpdateException`) maps to `Locked<Object>Exception` as we might want to do a retry when the object in no longer locked, but for the others we just use `DbUpdateException` as we are not treating them any different in terms of logic or logging.